### DISCUSSION TITLE
PHP Parse error during install

### DISCUSSION THREAD

**Discussion by ArachneNet** [kimai/kimai#3711] [On-Premise: Installation & Updates]
Hi, 
I hope this is the right place to ask this question. During the install process, which has been quite arduous to say the least. I arrive at the step where you run the kimai installer to install the local hosted server on linux using

`sudo bin/console kimai:install -n`

Sadly, this throws the error: `PHP Parse error:  syntax error, unexpected token "match" in /var/www/html/kimai2/vendor/willdurand/negotiation/src/Negotiation/Negotiator.php on line 41
`

This line says the following:
 `            return new Match($accept->getQuality() * $priority->getQuality(), $score, $index);
`

What I suspect has happened is that I have installed a different version of PHP. If I am not mistaken I am running 8.1. I am not a programmer nor an actual sysadmin so I probably made a dumb mistake somewhere. Any help at all would be really appreciated! Cheers!

**Comment by kevinpapst** [comment]
> which has been quite arduous to say the least

Oh no, I am soooo sorry to hear that. So I wasted thousands of hours creating a professional grade software which is for-free just to leave you with such a bad experience 😢 /sarcasm-off 

This is not my problem and you should stop posting stuff like that when actually asking for my time to help you.

> If I am not mistaken I am running 8.1.

Use `php -v` to verify. I assume you did some really weird things during the installation and now you are asking way too late.

**Comment by ArachneNet** [comment]
Update: 

After some more digging it seems to be an incompatability with PHP 8.*. I rnamed some of the match instances to my_match to resolve it but now it halts on the unexpected token "readonly",
`PHP Parse error:  syntax error, unexpected token "readonly" in /var/www/html/kimai2/vendor/jms/serializer/src/Metadata/Driver/AnnotationDriver.php on line 105
`
I suppose I'll uninstall it all and install with PHP 7.*.

That being said, I think you misunderstand @kevinpapst . I am not crying about it being hard to install. Hats off to you for creating such an extensive tool and then releasing it for free. It was a nod of humor because I suspect most people will have had an experience like I am having now where you try to install something and keep running into walls for three days straight. I mostly wondered if it could be confirmed that the kimai install is incompatible with PHP 8.* so I know where to look next. This github page is the closest I could find to a community help page. That's all.

**Reply by kevinpapst** [reply]
If you want to know compatibility, it is written in several places, e.g. https://github.com/kimai/kimai#requirements
